src/Service.Tests/Configuration/HotReload/ConfigurationHotReloadTests.cs

```diff
@@ -299,4 +299,41 @@ public async Task HotReloadConfigRuntimeGQLEnabledEndToEndTest()
         // Assert
         Assert.AreEqual(HttpStatusCode.NotFound, gQLResult.StatusCode);
     }
+
+    /// <summary>
+    /// Hot reload the configuration file by saving a new database type and connection string.
+    /// Validate that the response from the server is correct when making a new request after
+    /// the change in database type.
+    /// </summary>
+    [DataTestMethod]
+    [DataRow(DatabaseType.MySQL, TestCategory.MYSQL, DisplayName = "Hot-reload change to MySQL Datasource")]
+    [DataRow(DatabaseType.PostgreSQL, TestCategory.POSTGRESQL, DisplayName = "Hot-reload change to PostgreSQL Datasource")]
+    public async Task HotReloadConfigDatasourceDatabaseTypesEndToEndTest(DatabaseType expectedDatabaseType, string expectedTestCategory)
+    {
+        // Arrange
+        RuntimeConfig previousRuntimeConfig = _configProvider.GetConfig();
+        DatabaseType previousDatabaseType = previousRuntimeConfig.DataSource.DatabaseType;
+        string expectedConnectionString = $"{ConfigurationTests.GetConnectionStringFromEnvironmentConfig(expectedTestCategory).Replace("\\", "\\\\")}";
+
+        // Act
+        GenerateConfigFile(
+            databaseType: expectedDatabaseType,
+            connectionString: expectedConnectionString);
+        System.Threading.Thread.Sleep(3000);
+
+        // Await allows the server to hot-reload before sending the message since it has to wait
+        // until the hot-reload is finished before being able to handle any requests.
+        JsonElement reloadGQLContents = await GraphQLRequestExecutor.PostGraphQLRequestAsync(
+            _testClient,
+            _configProvider,
+            GQL_QUERY_NAME,
+            GQL_QUERY);
+        RuntimeConfig updatedRuntimeConfig = _configProvider.GetConfig();
+        DatabaseType actualDatabaseType = updatedRuntimeConfig.DataSource.DatabaseType;
+
+        // Assert
+        Assert.AreNotEqual(previousDatabaseType, actualDatabaseType);
+        Assert.AreEqual(expectedDatabaseType, actualDatabaseType);
+        SqlTestHelper.PerformTestEqualJsonStrings(_bookDBOContents, reloadGQLContents.GetProperty("items").ToString());
```

Comment:
_bookDBOContents has values from MSSQL, currently all the DBs have the same values, so we don't know if it actually picked the value from MSSQL or the updated DB. Would you be able to add one row to each db with different values which we can use for more robust testing?

Reply:
Also Based on how our pipelines work, we do not support cross database access in single pipeline. But you can add another test which i mentioned in this issue itself.